pkg/generic/thrift/http.go

```diff
@@ -20,33 +20,68 @@ import (
 	"context"
 
 	"github.com/apache/thrift/lib/go/thrift"
+	"github.com/bytedance/sonic"
+	"github.com/cloudwego/dynamicgo/conv"
+	"github.com/cloudwego/dynamicgo/conv/t2j"
 
 	"github.com/cloudwego/kitex/pkg/generic/descriptor"
+	"github.com/cloudwego/kitex/pkg/protocol/bthrift"
+	"github.com/cloudwego/kitex/pkg/remote"
+	"github.com/cloudwego/kitex/pkg/remote/codec/perrors"
+	cthrift "github.com/cloudwego/kitex/pkg/remote/codec/thrift"
 )
 
 // WriteHTTPRequest implement of MessageWriter
 type WriteHTTPRequest struct {
 	svc              *descriptor.ServiceDescriptor
 	binaryWithBase64 bool
+	enableDynamicgo  bool
+	opts             conv.Options
+	method           string
 }
 
-var _ MessageWriter = (*WriteHTTPRequest)(nil)
+var (
+	_          MessageWriter = (*WriteHTTPRequest)(nil)
+	customJson               = sonic.Config{
+		EscapeHTML: true,
+		UseNumber:  true,
+	}.Froze()
+)
 
 // NewWriteHTTPRequest ...
 // Base64 decoding for binary is enabled by default.
 func NewWriteHTTPRequest(svc *descriptor.ServiceDescriptor) *WriteHTTPRequest {
-	return &WriteHTTPRequest{svc, true}
+	return &WriteHTTPRequest{svc, true, false, conv.Options{}, ""}
 }
 
-// SetBase64Binary enable/disable Base64 decoding for binary.
+// SetBinaryWithBase64 enable/disable Base64 decoding for binary.
 // Note that this method is not concurrent-safe.
 func (w *WriteHTTPRequest) SetBinaryWithBase64(enable bool) {
 	w.binaryWithBase64 = enable
 }
 
-// Write ...
-func (w *WriteHTTPRequest) Write(ctx context.Context, out thrift.TProtocol, msg interface{}, requestBase *Base) error {
+// SetEnableDynamicgo enable/disable dynamicgo encoding.
+func (w *WriteHTTPRequest) SetEnableDynamicgo(enable bool) {
+	w.enableDynamicgo = enable
+}
+
+// SetConvOptions set the options to be used for dynamicgo encoding.
+func (w *WriteHTTPRequest) SetConvOptions(opts conv.Options) {
+	w.opts = opts
+}
+
+// SetMethodName set the method name.
+func (w *WriteHTTPRequest) SetMethodName(method string) {
+	w.method = method
+}
+
+func (w *WriteHTTPRequest) originalWrite(ctx context.Context, out thrift.TProtocol, msg interface{}, requestBase *Base) error {
 	req := msg.(*descriptor.HTTPRequest)
+	if req.Body == nil && len(req.RawBody) != 0 {
+		if err := customJson.Unmarshal(req.RawBody, &req.Body); err != nil {
+			return err
+		}
+	}
 	fn, err := w.svc.Router.Lookup(req)
```

Comment:
It looks like this code is duplicated with GetMethod defined in pkg/generic/httpthrift_codec.go.

Reply:
@jayantxie Actually this is an original code, and getMethod defined in pkg/generic/httpthrift_codec.go doesn't return function descriptor, so I don't think this code is a duplicate.

Reply:
All right.